src/include/duckdb/main/query_profiler.hpp

```diff
@@ -16,20 +16,41 @@
 #include "duckdb/common/unordered_map.hpp"
 #include "duckdb/common/winapi.hpp"
 #include "duckdb/execution/physical_operator.hpp"
-
+#include "duckdb/execution/expression_executor_state.hpp"
 #include <stack>
 #include <unordered_map>
 
 namespace duckdb {
+class ExpressionExecutor;
 class PhysicalOperator;
 class SQLStatement;
 
+struct ExpressionExecutionInformation {
+	explicit ExpressionExecutionInformation(ExpressionExecutor &executor);
+
+	//! Count the number of time the executor called
+	uint64_t total_count = 0;
+	//! Count the number of time the executor called since last sampling
+	uint64_t current_count = 0;
+	//! Count the number of samples
+	uint64_t sample_count = 0;
+	//! Count the number of tuples in all samples
+	uint64_t sample_tuples_count = 0;
+	//! Count the number of tuples processed by this executor
+	uint64_t tuples_count = 0;
+
+	vector<shared_ptr<ExpressionExecutorState>> states;
```

Comment:
Rather than having the `ExpressionExecutorState` here, I would prefer if the profiler had their own expression state class. This avoids having to drag the internal components of the expression executor into the query profiler, and also avoids having to turn this into shared pointers.